mkosi

```diff
@@ -2640,6 +2667,10 @@ def run_build_script(args, workspace, raw):
         else:
             cmdline.append("--chdir=/root")
 
+        if args.build_dir is not None:
+            cmdline.append("--setenv=BUILDDIR=/root/build")
```

Comment:
> i wanted that scripts can use the fact whether $BUILDDIR is set or not to decide whether they want to do an in-tree or out-of-tree build, 

That can still be decided by `[ "`realpath $BUILDDIR`" = "`realpath .`" ]`.

> I'd claim that removing mkosi.builddir/ is fine if it has the same effects as removing mkosi.default: i.e. break stuff...

Right, but then you have to make sure the dir always exist in your git repo via a .gitkeep file or such, and gitignore the rest. If BUILDDIR were always set, builds would just not be cached.
It's not the end of the world, but it is an annoyance.